napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -195,6 +210,209 @@ def open_perspective_popup(self):
+            + 'all the layers at the appropriate stride. '
+            + '0 is not a valid entry.'
+        )
+
+        stride_help_msg = trans._(
+            'Number of layers to place in each grid square before moving on to '
+            + 'the next square. The default ordering is to place the most visible '
+            + 'layer in the top left corner of the grid. A negative stride will '
+            + 'cause the order in which the layers are placed in the grid to be '
+            + 'reversed. '
+            + '0 is not a valid entry.'
+        )
+
+        # set up
+        wh_values = list(range(-1, 10))
+        wh_values = wh_values.pop(1)
+        grid_stride.setObjectName("gridStrideBox")
+        grid_stride.setAlignment(Qt.AlignCenter)
+        grid_stride.setMaximum(20)
+        grid_stride.setMinimum(-20)
+        grid_stride.setValue(self.viewer.grid.stride)
+        self.last_stride_value = (
+            self.viewer.grid.stride
+        )  # needed for skipping 0
+        grid_stride.valueChanged.connect(self._update_grid_stride)
+        self.grid_stride_box = grid_stride
+
+        grid_width.setObjectName("gridWidthBox")
+        grid_width.setAlignment(Qt.AlignCenter)
+        grid_width.setMaximum(20)
+        grid_width.setMinimum(-1)
+        grid_width.setValue(self.viewer.grid.shape[1])
+        self.last_width_value = self.viewer.grid.shape[
+            1
+        ]  # needed for skipping 0
+        grid_width.valueChanged.connect(self._update_grid_width)
+        self.grid_width_box = grid_width
+
+        grid_height.setObjectName("gridStrideBox")
+        grid_height.setAlignment(Qt.AlignCenter)
+        grid_height.setMaximum(20)
+        grid_height.setMinimum(-1)
+        grid_height.setValue(self.viewer.grid.shape[0])
+        self.last_height_value = self.viewer.grid.shape[
+            0
+        ]  # needed for skipping 0
+        grid_height.valueChanged.connect(self._update_grid_height)
+        self.grid_height_box = grid_height
+
+        # The following is needed in order to make the tooltip wrap the text.
+        shape_help_txt = f"<FONT> {shape_help_msg}</FONT>"
+        stride_help_txt = f"<FONT> {stride_help_msg}</FONT>"
+
+        shape_help_symbol.setObjectName("help_label")
+        shape_help_symbol.setToolTip(shape_help_txt)
+
+        stride_help_symbol.setObjectName("help_label")
+        stride_help_symbol.setToolTip(stride_help_txt)
+
+        # layout
+        form_layout = QFormLayout()
+
+        form_layout.insertRow(
+            0,
+            QLabel(trans._('Grid stride:'), parent=popup),
+            grid_stride,
+        )
+
+        form_layout.insertRow(
+            1,
+            QLabel(trans._('Grid width:'), parent=popup),
+            grid_width,
+        )
+
+        form_layout.insertRow(
+            2,
+            QLabel(trans._('Grid height:'), parent=popup),
+            grid_height,
+        )
```

Comment:
```suggestion
        form_layout.insertRow(0, QLabel(trans._('Grid stride:')), grid_stride)
        form_layout.insertRow(1, QLabel(trans._('Grid width:')), grid_width)
        form_layout.insertRow(2, QLabel(trans._('Grid height:')), grid_height)
```

I'm not sure these parent's are doing anything, since adding a widget to a layout reparents it.
https://doc.qt.io/qt-5/layout.html#tips-for-using-layouts